framework/src/modules/poa/module.ts

```diff
@@ -206,23 +212,18 @@ export class PoAModule extends BaseModule {
 			throw new Error('`address` property of all entries in validators must be distinct.');
 		}
 
-		const sortedValidatorsByAddress = [...validatorAddresses].sort((a, b) => a.compare(b));
-		for (let i = 0; i < validators.length; i += 1) {
-			// Check that entries in the validators array are ordered lexicographically according to address.
-			if (!validatorAddresses[i].equals(sortedValidatorsByAddress[i])) {
-				throw new Error('`validators` must be ordered lexicographically by address.');
-			}
+		if (!objects.isBufferArrayOrdered(validatorAddresses)) {
+			throw new Error('`validators` must be ordered lexicographically by address.');
+		}
 
-			if (!/^[a-z0-9!@$&_.]+$/g.test(validators[i].name)) {
+		for (const poaValidator of validators) {
+			if (!/^[a-z0-9!@$&_.]+$/g.test(poaValidator.name)) {
```

Comment:
if there is a constant for this regex then use that or else create a new constant and use it here